cmd/release-controller/bugzilla.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// syncBugzilla checks whether fixed bugs in a release had their
+// PR reviewed and approved by the QA contact for the bug
+func (c *Controller) syncBugzilla(key queueKey) error {
+	defer func() {
+		err := recover()
+		panic(err)
+	}()
+
+	release, err := c.loadReleaseForSync(key.namespace, key.name)
+	if err != nil || release == nil {
+		return err
+	}
+
+	glog.V(4).Infof("Verifying fixed bugs %s", release.Config.Name)
+
+	// check if verifyBugs publish step is enabled for release
+	verifyBugs, ok := release.Config.Publish[publishTypeBugzilla]
+	if !ok || verifyBugs.Disabled {
+		return nil
+	}
+
+	// get accepted tags
+	acceptedTags := sortedRawReleaseTags(release, releasePhaseAccepted)
+	if len(acceptedTags) == 0 {
+		return nil
+	}
+	var errs []error
+	// get latest accepted tag
+	tag := acceptedTags[0]
+	// check if tag has already been bugzilla verified
+	if tag.Annotations[releaseAnnotationBugsVerified] == "true" {
+		return nil
+	}
+
+	verifyComplete := false
+	info, err := c.getReleaseTagInfo(release.Config.Name, tag.Name, "")
+	if err != nil {
+		glog.Errorf("Failed to get release tag info for %s: %v", tag, err)
+	} else if info.Previous != nil {
+		if bugs, err := c.releaseInfo.Bugs(info.Previous.Name, info.Tag.Name); err != nil {
+			errs = append(errs, fmt.Errorf("Unable to generate changelog from %s to %s: %v", info.Previous.Name, info.Tag.Name, err))
+		} else {
+			errs = append(errs, c.bugzillaVerifier.VerifyBugs(strings.Split(bugs, "\n"))...)
+			// we may want to not mark as completed if errors occurred; do so for now to prevent rerunning
+			// and failing mutiple times
+			verifyComplete = true
+		}
+	} else {
+		glog.V(4).Infof("No previous version found for tag %s", info.Tag.Name)
+	}
+
+	if verifyComplete {
+		if tag.Annotations == nil {
```

Comment:
You cannot modify the image stream in place.  You have to do a deep copy, then edit (otherwise you will cause races between this loop and the main loop).  See how the sync controller works.